examples/asset/arc_asset.rs

```diff
@@ -0,0 +1,251 @@
+//! This example illustrates how to use assets in an async context (through locking).
```

Comment:
I dropped the whole "locking" wording, that was actually just an artifact from my "asset locking" PR (which I stole this example from, lol).

Reply:
I still do like the naming of "locking" an asset, since that is effectively what you're doing when you've got 2+ ref counts to an asset - you're locking it for mutation. Maybe this term could be used in docs somewhere?
```
/// When an asset has a reference count of more than 1, this asset is considered *locked* for mutation.
```